python/paddle/fluid/dataloader/batch_sampler.py

```diff
@@ -0,0 +1,125 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+from __future__ import division
+
+import numpy as np
+
+__all__ = ["BatchSampler"]
+
+
+class BatchSampler(object):
+    """
+    A base implement of batch sampler used by `fluid.io.DataLoader`
+    which yield mini-batch indices(a list/tuple with length as
+    mini-batch size and holds sample indices) iterably.
+
+    Batch sampler used by `fluid.io.DataLoader` should be a subclass
+    of `fluid.io.BatchSampler`, BatchSampler subclasses should
+    implement following methods:
+
+    :math:`__iter__`: return mini-batch indices iterably.
+    :math:`__len__`: get mini-batch number in an epoch.
+
+
+    Args:
+        data_source(Dataset): this could be a `fluid.io.Dataset` 
+                implement or other python object which implemented
+                `__len__` for BatchSampler to get indices as the
+                range of :attr:`data_source` length. Default None.
+        indices (list|tuple): a substitution parameter for
+                :attr:`data_source` either :attr:`data_source` or
+                :attr:`indices` should be set, give the whole
+                indices to sampler from directly. Default None.
+        shuffle(bool): whther to shuffle indices order before genrate
+                batch indices. :attr:`shuffle` should not be set if 
+                :attr:`sample_iter` is set. Default False.
```

Comment:
1. genrate -> generating.
2. sample_iter is not in init func, but used in  TestBatchSamplerWithIndicesAndDataSource

Reply:
Done, thanks!